cmd/gke-certificates-controller/app/gke_approver.go

```diff
@@ -259,21 +269,15 @@ func validateNodeServerCert(opts approverOptions, csr *capi.CertificateSigningRe
 	return true
 }
 
-// Maximum number of SANs in server CSR.
-const maxServerSANs = 10
-
-// Only check that DNS name and IP in SAN match an existing VM in the project.
+// Only check that IPs in SAN match an existing VM in the project.
 // Username was already checked against CN, so this CSR is coming from
 // authenticated kubelet.
 func validateNodeServerCertInner(client *http.Client, opts approverOptions, csr *capi.CertificateSigningRequest, x509cr *x509.CertificateRequest) error {
-	numDNS, numIPs := len(x509cr.DNSNames), len(x509cr.IPAddresses)
 	switch {
-	case numDNS == 0 || numIPs == 0:
-		return errors.New("no SAN DNS names or IPs")
-	case numDNS+numIPs > maxServerSANs:
-		return fmt.Errorf("got %d SANs, want at most %d", numDNS+numIPs, maxServerSANs)
+	case len(x509cr.IPAddresses) == 0:
+		return errors.New("no SAN IPs")
 	case len(x509cr.EmailAddresses) > 0 || len(x509cr.URIs) > 0:
-		return errors.New("only DNS and IP SANs allowed")
+		return errors.New("only IP SANs allowed")
```

Comment:
Do we know why kubelet with the gcp cloud provider doesn't add any dns names today? I want to make sure that a change in the GCE API won't break this suddenly.

Reply:
GCE cloud provider only provides IPs: https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/gce/gce_instances.go#L84

Should it be adding instance name as DNS name? It's technically not a resolvable DNS name, so I'd assume that wouldn't be correct anyway.

Reply:
I can resolve node names from the master:
```
root@k-6-master:~# nslookup k-6-minion-group-16zh
Server:         169.254.169.254
Address:        169.254.169.254#53

Non-authoritative answer:
Name:   k-6-minion-group-16zh.c.mikedanese-k8s.internal
Address: 10.240.0.3
```

Reply:
Also see https://cloud.google.com/compute/docs/internal-dns#instance-fully-qualified-domain-names

Reply:
ack, i guess that works within GCP.
The check above doesn't prevent the DNS SANs, but also doesn't validate them.
I can change the gce cloud provider and add DNS validation here later, wdyt?

Reply:
sgtm